web/bfex/components/data_pipeline/tasks/researchid.py

```diff
@@ -34,60 +28,54 @@ def run(self, data):
         :return: last faculty member handled
         """
         
-        no_text_count = 0
-        for faculty in data:
+        faculty = data
+        if isinstance(faculty, str):
+            faculty_name = faculty
+        else:
             faculty_name = faculty.name
-    
-            search_results = Faculty.search().query('match', name=faculty_name).execute()
-            if len(search_results) > 1:
-                # Shouldn't happen, but could.
-                raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
 
-            search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="ResearchId")
-            search_dup.delete()
-            faculty = search_results[0]
-            if faculty.research_id is not None:
-                
-                scraper = ScraperFactory.create_scraper(faculty.research_id, ScraperType.RESEARCHID)
-                scrapps = scraper.get_scrapps()
+        search_results = Faculty.search().query('match', name=faculty_name).execute()
+        if len(search_results) > 1:
+            # Shouldn't happen, but could.
+            raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
+
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="ResearchId")
+        search_dup.delete()
```

Comment:
Are you using this later? Does it need to be stored? You can just put Document...delete()

Reply:
pep8 line too long?

Reply:
You can break up lines on the . so:

Document.search()
                 .query(...)
                 .query()

etc.

Reply:
kek